src/detect-engine-address.c

```diff
@@ -892,6 +892,18 @@ int DetectAddressSetup(DetectAddressHead *gh, char *s)
 }
 
 /**
+ * A list of variables we try to resolve.
+ * Used by DetectAddressParse2() to detect loops in variables declaration
+ */
+typedef struct ResolvedVariable_ {
+    char *var_name;
+    TAILQ_ENTRY(ResolvedVariable_) next;
+} ResolvedVariable;
+
+static TAILQ_HEAD(, ResolvedVariable_) resolved_variables_list =
```

Comment:
if no one outside of DetectAddressParse2 accesses this list, can we move it into DetectAddressParse2? Saves us a global.